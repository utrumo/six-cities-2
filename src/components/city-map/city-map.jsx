import React from 'react';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';
import Selectors from '../../store/selectors.js';

import Map from '../map/map.jsx';

const CitiyMap = ({view, markers, activeCardId}) => (
  <section className="cities__map map">
    <Map
      view={view}
      markers={markers}
      activeCardId={activeCardId}
    />
  </section>
);

CitiyMap.propTypes = {
  activeCardId: PropTypes.number.isRequired,

  view: PropTypes.exact({
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired,
    zoom: PropTypes.number.isRequired
  }),

  markers: PropTypes.arrayOf(PropTypes.exact({
    id: PropTypes.number.isRequired,
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired
  }))
};

const mapStateToProps = (state) => ({
  view: Selectors.getFirstOfferCityLocation(state),
  markers: Selectors.getCityMapMarkers(state),
  activeCardId: Selectors.getCurrentOfferId(state)
});

export {CitiyMap};
export default connect(mapStateToProps)(CitiyMap);
