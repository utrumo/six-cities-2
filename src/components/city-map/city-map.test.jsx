import React from 'react';
import renderer from 'react-test-renderer';
import {CitiyMap} from './city-map.jsx';

const view = {
  latitude: 52.370216,
  longitude: 4.895168,
  zoom: 12
};

const markers = [
  {
    id: 10,
    latitude: 52.3909553943508,
    longitude: 4.85309666406198
  },
  {
    id: 11,
    latitude: 52.369553943508,
    longitude: 4.85309666406198
  }
];

it(`CityMap renders correctly`, () => {
  const tree = renderer.create(<CitiyMap view={view} markers={markers} activeCardId={1}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
