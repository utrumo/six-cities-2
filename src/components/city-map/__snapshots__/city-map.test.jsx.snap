// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityMap renders correctly 1`] = `
<section
  className="cities__map map"
>
  <div
    id="map"
    style={
      Object {
        "height": "100%",
      }
    }
  />
</section>
`;
